router/mocks_test.go

```diff
@@ -0,0 +1,108 @@
+package router
+
+import (
+	wt "github.com/zettio/weave/testing"
+	"net"
+	"testing"
+)
+
+type peerQueue struct {
+	peers []*Peer
+}
+
+func (q *peerQueue) clear() {
+	q.peers = nil
+}
+
+// Construct a Router object with a mock interface to check peer
+// garbage-collection, and without firing up any ancilliary goroutines
+func NewTestRouter(t *testing.T, name PeerName, queue *peerQueue) *Router {
+	onMacExpiry := func(mac net.HardwareAddr, peer *Peer) {}
+	onPeerGC := func(peer *Peer) {
+		//t.Log("Removing unreachable", peer)
+		if queue != nil {
+			queue.peers = append(queue.peers, peer)
+		}
+	}
+	router := newRouter(nil, name, nil, 10, 1024, nil, onMacExpiry, onPeerGC)
+	router.ConnectionMaker = &ConnectionMaker{
+		ourself:   router.Ourself,
+		queryChan: make(chan *ConnectionMakerInteraction, ChannelSize)}
+	router.Routes = NewRoutes(router.Ourself.Peer, router.Peers)
+	router.Routes.queryChan = make(chan *Interaction, ChannelSize)
+	return router
+}
+
+func (r1 *Router) AddTestConnection(r2 *Router) {
+	toName := r2.Ourself.Peer.Name
+	toPeer := NewPeer(toName, r2.Ourself.Peer.UID, 0)
+	r1.Peers.FetchWithDefault(toPeer) // Has side-effect of incrementing refcount
+	r1.Ourself.Peer.connections[toName] = &mockConnection{toPeer, ""}
+	r1.Ourself.Peer.version += 1
+}
+
+func (r0 *Router) AddTestRemoteConnection(r1, r2 *Router) {
+	fromName := r2.Ourself.Peer.Name
+	fromPeer := NewPeer(fromName, r1.Ourself.Peer.UID, 0)
+	fromPeer = r0.Peers.FetchWithDefault(fromPeer)
+	toName := r2.Ourself.Peer.Name
+	toPeer := NewPeer(toName, r2.Ourself.Peer.UID, 0)
+	toPeer = r0.Peers.FetchWithDefault(toPeer)
+	r0.Ourself.Peer.connections[toName] = &RemoteConnection{fromPeer, toPeer, ""}
+	r0.Ourself.Peer.version += 1
+}
+
+func (r1 *Router) DeleteTestConnection(r2 *Router) {
+	toName := r2.Ourself.Peer.Name
+	toPeer, _ := r1.Peers.Fetch(toName)
+	toPeer.DecrementLocalRefCount()
+	delete(r1.Ourself.Peer.connections, toName)
+	r1.Ourself.Peer.version += 1
+}
+
```

Comment:
I've [just added](https://github.com/zettio/weave/commit/1a31748e4092b3519a58e8b87ee0996a5f4509f5) some helper functions to `LocalPeer` which can be called instead of directly modifying `connections` and `versions` in all of the above.